pycontrails/models/dry_advection.py

```diff
@@ -0,0 +1,430 @@
+"""Simulate dry advection."""
+
+from __future__ import annotations
+
+import dataclasses
+from typing import Any, NoReturn, overload
+
+import numpy as np
+
+from pycontrails.core import models
+from pycontrails.core.flight import Flight
+from pycontrails.core.met import MetDataset
+from pycontrails.core.met_var import AirTemperature, EastwardWind, NorthwardWind, VerticalVelocity
+from pycontrails.core.vector import GeoVectorDataset
+from pycontrails.models.cocip import contrail_properties, wind_shear
+from pycontrails.physics import geo, thermo
+
+
+@dataclasses.dataclass
+class DryAdvectionParams(models.ModelParams):
+    """Parameters for the :class:`DryAdvection` model."""
+
+    #: Apply Euler's method with a fixed step size of ``dt_integration``. Advected waypoints
+    #: are interpolated against met data once each ``dt_integration``.
+    dt_integration: np.timedelta64 = np.timedelta64(30, "m")
+
+    #: Max age of contrail evolution.
+    max_age: np.timedelta64 = np.timedelta64(20, "h")
+
+    #: Difference in altitude between top and bottom layer for stratification calculations,
+    #: [:math:`m`]. Used to approximate derivative of "lagrangian_tendency_of_air_pressure" layer.
+    dz_m: float = 200.0
+
+    #: Upper bound for evolved plume depth, constraining it to realistic values.
+    max_depth: float | None = 1500.0
+
+    #: Initial plume width, [:math:`m`]. Overridden by "width" key on :attr:`source`.
+    # If None, only pointwise advection is simulated without wind shear effects.
+    width: float | None = 100.0
+
+    #: Initial plume depth, [:math:`m`]. Overridden by "depth" key on :attr:`source`.
+    # If None, only pointwise advection is simulated without wind shear effects.
+    depth: float | None = 100.0
+
+    #: Initial plume direction, [:math:`m`]. Overridden by "azimuth" key on :attr:`source`.
+    # If None, only pointwise advection is simulated without wind shear effects.
+    azimuth: float | None = 0.0
+
+
+class DryAdvection(models.Model):
+    """Simulate "dry advection" of a plume with an elliptical cross section.
```

Comment:
"...of an **emissions** plume"

Reply:
47fdf46dfb902a208650e13d6a0ac6f5ebb23b25